[Ong Kan Wu] iP

OKW32 opened the pull request.

OKW32 committed:
Initialize: Create skeletal code for Beefy.java

OKW32 committed:
Style: Create Main class and add empty line at end of Beefy.java

Beefy.java is the entry point.

Creating a Main class to be used as the entry point for our program is
cleaner and follows the coding standards.

Let's
* Create a new class Main
* Create class functions in Beefy e.g printSeparation(), startChat(),
endChat() to be used in Main class

OKW32 committed:
Feature: Add echo user command

OKW32 committed:
Feature: Add add and list functionality

OKW32 committed:
Feature: Add ability to mark and unmark tasks

All text typed by user is stored in Beefy class.

Creating two new classes to execute the tasks of storing information of
the task and storage of all tasks created will make the code easier
to understand.

Let's
* Create a Task class to store information of each individual task
* Create a TaskList class to compile and store all tasks that have been
created

OKW32 committed:
Style: Shift creation of Task variable to addTask function

Task is created in startChat() function.

Shifting the creation of Task variable from startChat function in
Beefy class to the addTask function in TaskList class, will follow the
proper coding standards.

Let's change the function addTask in TaskList to accept the task
description and create the Task variable in it.

OKW32 committed:
Style: Change constant name from BOTNAME to BOT_NAME

OKW32 committed:
Feature: Add ToDos, Events and Deadline functionalities

Let's
* Define an enum type, CommandType, to help determine which command
should be executed
* Create 3 new children classes of Task (ToDo, Deadline, Event) to
allow for differentiation between the types of task to add to
TaskList
* Create a function, executeCommand, to receive user's input and
command type so as to determine which command to execute

OKW32 committed:
Refactor code: Change executeCommand + Add space for task printing

executeCommand function does not handle cases where user input is
formatted incorrectly for the adding of Tasks.

Checking for above mentioned cases would prevent the code from
exiting with error.

Let's add try and catch statements to print the proper format to use
for users to view, upon detection of improper formatting.

OKW32 committed:
Test: Add TextUiTesting + Refactor startChat

startChat function unable to print out to user the proper formatting
for mark/unmark task upon improper input formatting.

Let's
* Use a nested if else statement instead to check if the input after
mark/unmark command is an integer
* Add inputs in input.txt to test for all features of ToDos,
Deadlines, Events, List, Mark and Unmark, as well as to test if
exceptions for isInteger and handleCommand functions are executed
correctly.

ningsongshen commented on src/main/java/Beefy.java:
```diff
@@ -0,0 +1,121 @@
+    }
+
+    private void printSeparation() {
+        int WIDTH = 59;
```
great use of slap

ningsongshen commented on src/main/java/Beefy.java:
```diff
@@ -0,0 +1,121 @@
+        printSeparation();
+        System.out.println(BOT_NAME);
+        System.out.println("G'Day! I'm " + BOT_NAME);
+        System.out.println("What can I do for you mate?");
```
Non-american slang!

ningsongshen commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,119 @@
+____________________________________________________________
+BEEFY
+Are you blind mate?
+____________________________________________________________
```
lol

ningsongshen commented on src/main/java/Beefy.java:
```diff
@@ -0,0 +1,121 @@
+            } else if (userWords.length == 2 && userWords[0].equalsIgnoreCase("unmark")) {
+                if (isInteger(userWords[1])) {
+                    executeCommand(userWords[1], CommandType.UNMARK);
+                } else {
```
many if statements, i see why needed, but maybe use a case?

ningsongshen commented on src/main/java/Beefy.java:
```diff
@@ -0,0 +1,121 @@
+            break;
+        case DEADLINE:
+            taskDetails = userWords.trim().split("/by");
+            try {
```
So far we assume the user to ahve perfect input

ningsongshen reviewed (changes requested):
Great work friend!

Leong-ZM commented on src/main/java/Beefy.java:
```diff
@@ -0,0 +1,121 @@
+
+    public void endChat() {
+        System.out.println(
+
```
unnecessary blank lines?

Leong-ZM commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,7 @@
+public class Main {
+    public static void main(String[] args) {
+        Beefy chatBot = new Beefy();
```
good use of abstraction

Leong-ZM commented on src/main/java/Beefy.java:
```diff
@@ -0,0 +1,121 @@
+import java.util.Scanner;
```
good explicit import

Leong-ZM commented on src/main/java/Beefy.java:
```diff
@@ -0,0 +1,121 @@
+                if (isInteger(userWords[1])) {
+                    executeCommand(userWords[1], CommandType.MARK);
+                } else {
+                    System.out.println("Don't be a Drongo mate. Use format:"
```
good line length

Leong-ZM reviewed (commented):
generally good code standard

OKW32 committed:
Style: Remove unnecessary lines in endChat function

Joellimjr commented on src/main/java/CommandType.java:
```diff
@@ -0,0 +1,3 @@
+public enum CommandType {
```
Great way to organise the different commands and use what we are learning

Joellimjr commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+        System.out.println("---Number of Tasks in List: " + numberOfTasks + "---");
+    }
+
+    public void addTask(String taskDescription, String from, String to) {
```
Great use of overloading

Joellimjr commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,119 @@
+You
+____________________________________________________________
+BEEFY
+Don't be a Drongo mate. Use format:
```
lol

Joellimjr commented on src/main/java/Beefy.java:
```diff
@@ -0,0 +1,115 @@
+            printSeparation();
+            String[] userWords = userLine.trim().split("\\s+", 2);
+            System.out.println(BOT_NAME);
+            if (userLine.equalsIgnoreCase("bye")) {
```
perhaps have a separate function that handles invalid formats to make this portion of code a bit easier to read

Joellimjr commented on src/main/java/Beefy.java:
```diff
@@ -0,0 +1,115 @@
+            try {
+                userTasks.addTask(taskDetails[0].trim(), taskDetails[1].trim(), taskDetails[2].trim());
+            } catch (ArrayIndexOutOfBoundsException e) {
+                System.out.println("Don't be a Drongo mate. Use format:" + System.lineSeparator() + "event (Task Description) /from (Date) /to (Date)");
```
Good error handling! Helps users understand what to do

OKW32 committed:
Refactor code: Add command classes to improve code quality

All commands are handled inside Beefy, causing code to be
unreadable as there are multiple levels of abstraction.

Adding multiple command classes help to distribute the code out into
multiple classes and improves readability.

Let's create an interface Command, and create commands for each of
the different command type to implement the interface.

OKW32 committed:
Style: Add/Remove blank lines

OKW32 committed:
Merge branch 'branch-Level-5'

OKW32 committed:
Package: Create beefy, command, parser, task and ui packages

OKW32 committed:
Merge branch 'branch-A-Packages'

OKW32 committed:
Refactor code: Change error handling

Throw and catch statements with different exceptions are written in
the execute function in each individual command class, making it
hard to scale code in future.

Creating BeefyException class for all our execute functions and
class constructors to throw will allow for better scalability of
code in future.

OKW32 committed:
Feature: Add Delete functionality

Let's add deleteTask function in TaskList class, a switch statement
in Parser class to handle delete command, and a DeleteCommand class.

OKW32 committed:
Feature: Add Save functionality

Let's
* Create a Data folder to store all tasks added in a text file
(beefy.txt) in it
* Create a storage package with Storage class to handle all storage
functions such as converting beefy.txt into a TaskList, and updating
storage
* Modify TaskList.addTask function to not output anything during
initialization by adding a second Boolean variable initialize

OKW32 committed:
Merge branch 'branch-Level-6'

OKW32 committed:
Merge branch 'branch-Level-7'

OKW32 committed:
Feature: Implement save function after delete/mark/unmark task

Let's call the function updateDisk in the execute function for the
respective command classes.

OKW32 committed:
Bug Fix: Fix Storage class readDisk function

Does not create directory to store storage text file when there is no
directory found, causing IO exceptions to occur upon writing to disk.

Creating a file when there is no directory found will prevent IO
exceptions from occurring.

Let's create a function that will be called to create the directory and
file when there is a FileNotFoundException.

OKW32 committed:
Refactor code: Refactor Mark/Unmark task function

Marking/unmarking tasks during initialization will cause output to be
printed.

Checking whether we are marking/unmarking task in initialization phase
before deciding on the need for output will solve the issue.

Let's create a Boolean variable isInitialize to check if the function is
called in the initialization stage.

StanleyW00 commented on src/main/java/beefy/command/ByeCommand.java:
```diff
@@ -0,0 +1,22 @@
+        }
+    }
+
+    @Override
+    public void execute() {
+
+    }
```
Try to avoid empty methods. Perhaps, it can be filled with lines or removed.

StanleyW00 commented on src/main/java/beefy/command/DeleteCommand.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Override
+    public void execute() throws BeefyException, IOException {
+        if (taskId < 1 || taskId > userTasks.getNumberOfTasks())
+        {
+            throw new BeefyException("Can you not do math, mate?");
+        }
```
Try to use K&R style for if statements. Perhaps, move the brackets based on the coding standards.

StanleyW00 commented on src/main/java/beefy/storage/Storage.java:
```diff
@@ -0,0 +1,77 @@
+        fw.close();
+    }
+
+    public static void addToDisk(String taskLine) throws IOException{
```
Perhaps, there should be spacing between the exception and the bracket.

StanleyW00 reviewed (commented):
Overall, nice job in maintaining good code quality and following the coding standards. Just need to do some fixes.

OKW32 committed:
Style: Edit execute function in DeleteCommand class to follow K&R style

OKW32 committed:
Feature: Make chatbot able to understand dates and times

Chatbot can accept any string input for commands deadline and event.

Allowing chatbot to only accept a specific dateTime format for the
dateTime inputs will ensure proper deadline and event duration are
being stored. This ensures clean data storage.

Let's
* Refactor Deadline and Event class to store dateTime information in a
variable of type LocalDateTime
* Refactor DeadlineCommand and EventCommand constructors to throw an
exception when deadline's/event's dateTime information is not keyed in
using the proper dateTime format
* Refactor readDisk function in Storage class to parse dateTime
information stored in csv file into a LocalDateTime variable

OKW32 committed:
Merge pull request #1 from OKW32/branch-Level-8

Feature: Make chatbot able to understand dates and times

OKW32 committed:
Feature: Add Find functionality

Let's
* Create a Find Command class to handle the case when a find command is
to be executed
* Add a case statement to handle the find command in Parser class
* Add the function findTask in TaskList class to find all matching tasks
and print them out

OKW32 committed:
Merge branch 'master' into branch-Level-9

OKW32 committed:
Merge pull request #2 from OKW32/branch-Level-9

Branch level 9

OKW32 committed:
Documentation: Add JavaDoc comments to all non-trival code

OKW32 committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/beefy/task/TaskList.java

OKW32 committed:
Merge pull request #3 from OKW32/branch-A-JavaDoc

Branch a java doc

OKW32 committed:
Add User Guide

OKW32 committed:
Bug: Fix bug in message printing for marking and unmarking of tasks

Printing of message occurs during the initialization phase instead
of chat phase.

Let's change the if condition in both markTask and unmarkTask
function from (isInitialize) to (!isInitialize).

OKW32 committed:
Refactor code: Refactor code for adding of deadline/event

An empty task description can be added into the task list.

Making the code check if the task to be added has no description
will be able to ensure clean data storage.

Let's make the DeadlineCommand and EventCommand constructors throw
a BeefyException when an empty task description is passed into it.
